utils/src/main/java/io/atomix/utils/concurrent/SingleThreadContext.java

```diff
@@ -133,4 +168,25 @@ public void close() {
     executor.shutdownNow();
   }
 
+  class WrappedRunnable implements Runnable {
+
+    private final Runnable command;
+
+    WrappedRunnable(Runnable command) {
+      this.command = command;
+    }
+
+    @Override
+    public void run() {
+      try {
+        command.run();
+      } catch (Exception e) {
+        uncaughtExceptionObserver.accept(e);
+      } catch (Throwable e) {
```

Comment:
Do we need the other catch then?

Reply:
I am not sure if we should rethrow the exceptions. That's why I  handled them differently. What do you think?